src/spclient.rs

```diff
@@ -125,168 +312,61 @@ impl SpClient {
                     Receiver::new(0, scan_pubkey, spend_pubkey, change_label.into(), network)?;
             }
         }
-        let writer = FileWriter::new(path, label.clone())?;
 
         Ok(Self {
             label,
             scan_sk,
             spend_key,
             mnemonic,
             sp_receiver,
-            birthday,
-            last_scan: if birthday == 0 { 0 } else { birthday - 1 },
-            owned: HashMap::new(),
-            writer,
         })
     }
 
-    pub fn try_init_from_disk(label: String, path: String) -> Result<SpClient> {
-        let empty = SpClient::new(
-            label,
-            SecretKey::from_slice(&[1u8; SECRET_KEY_SIZE]).unwrap(),
-            SpendKey::Secret(SecretKey::from_slice(&[1u8; SECRET_KEY_SIZE]).unwrap()),
-            None,
-            0,
-            false,
-            path,
-        )?;
-
-        empty.retrieve_from_disk()
-    }
-
-    pub fn update_last_scan(&mut self, scan_height: u32) {
-        self.last_scan = scan_height;
-    }
-
-    pub fn get_spendable_amt(&self) -> u64 {
-        self.owned
-            .values()
-            .filter(|x| x.spend_status == OutputSpendStatus::Unspent)
-            .fold(0, |acc, x| acc + x.amount)
+    pub fn get_receiving_address(&self) -> String {
+        self.sp_receiver.get_receiving_address()
     }
 
-    #[allow(dead_code)]
-    pub fn get_unconfirmed_amt(&self) -> u64 {
-        self.owned
-            .values()
-            .filter(|x| match x.spend_status {
-                OutputSpendStatus::Spent(_) => true,
-                _ => false,
-            })
-            .fold(0, |acc, x| acc + x.amount)
+    pub fn get_scan_key(&self) -> SecretKey {
+        self.scan_sk
     }
 
-    pub fn extend_owned(&mut self, owned: Vec<(OutPoint, OwnedOutput)>) {
-        self.owned.extend(owned);
+    pub fn get_spend_key(&self) -> SpendKey {
+        self.spend_key.clone()
     }
 
-    pub fn check_outpoint_owned(&self, outpoint: OutPoint) -> bool {
-        self.owned.contains_key(&outpoint)
+    pub fn get_mnemonic(&self) -> Option<String> {
+        self.mnemonic.clone()
     }
 
-    pub fn mark_transaction_inputs_as_spent(&mut self, tx: Transaction) -> Result<()> {
-        let txid = tx.txid();
-
-        // note: this currently fails for collaborative transactions
-        for input in tx.input {
-            self.mark_outpoint_spent(input.previous_output, txid)?;
+    pub fn try_get_secret_spend_key(&self) -> Result<SecretKey> {
+        match self.spend_key {
+            SpendKey::Public(_) => Err(Error::msg("Don't have secret key")),
```

Comment:
Also not really related to this PR, but since sp_client is now a library and not an application, I would like to remove the `anyhow` errors and define proper errors like in `rust-silentpayments`. But that is a big change and deserves its own PR.

Reply:
Agree, but that's indeed a big change that needs to be dealt with on another PR